index.js

```diff
@@ -148,14 +148,19 @@ function getCombinedBranch(branchConfig) {
   return branchName;
 }
 
-async function handleSwitchToBranchCommand(sg, sortedBranches, combinedBranch) {
+async function handleSwitchToBranchCommand(sg, currentBranch, sortedBranches, combinedBranch) {
   const switchToBranchIndex = program.args[0];
   if (typeof switchToBranchIndex === 'undefined') {
     return;
   }
   let targetBranch;
+  let currentBranchIndex = sortedBranches.indexOf(currentBranch)
   if (switchToBranchIndex === 'combined') {
     targetBranch = combinedBranch;
+  } else if (switchToBranchIndex.startsWith("+")) {
+      targetBranch = sortedBranches[currentBranchIndex + parseInt(switchToBranchIndex.slice(1))]
+  } else if (switchToBranchIndex.startsWith("-")) {
+      targetBranch = sortedBranches[currentBranchIndex - parseInt(switchToBranchIndex.slice(1))]
```

Comment:
You have not accounted for index out of bounds errors because the provided argument is larger than the number of branches in the chain.
```suggestion
  } else if (switchToBranchIndex.startsWith("+")) {
      const nextIndex = Math.min(sortedBranches.length - 1, currentBranchIndex + parseInt(switchToBranchIndex.slice(1)));
      targetBranch = sortedBranches[nextIndex];
  } else if (switchToBranchIndex.startsWith("-")) {
      const nextIndex = Math.max(0, currentBranchIndex - parseInt(switchToBranchIndex.slice(1)));
      targetBranch = sortedBranches[nextIndex];
```